### ISSUE TITLE
Update Server UI Style Issue

### ISSUE THREAD

**Issue by keerthij** [eclipse/hawkbit#568]
When I tried to access update server UI from browser, I'm seeing the UI as in the image mentioned below. Looks like style issue. However, device simaulator UI is fine. Please find below the screenshots.

![image](https://user-images.githubusercontent.com/5600925/29258622-c899f8ee-80d8-11e7-8998-7f8e571b8b83.png)

![image](https://user-images.githubusercontent.com/5600925/29258956-f68850e0-80db-11e7-9028-0db1ff7e0210.png)

**Comment by michahirsch** [comment]
Hi @keerthij 

the hawkBit style and vaadin-widgetset is built in the `hawkbit-ui` module of the maven project. You should find the necessary generated `widgetset` and `style.css` in there: 
![image](https://user-images.githubusercontent.com/10206349/29259254-84a2a490-80c1-11e7-9ba2-1677896784d8.png)

Then you should also be able to access the style sheet and theme information by the link `http://localhost:8080/VAADIN/themes/hawkbit/styles.css?v=7.7.9`

You can also verfiy the maven build console output that the `vaadin-maven-plugin` was executed for the `hawkbit-ui` module
```
[INFO] --- vaadin-maven-plugin:7.7.9:resources (default) @ hawkbit-ui ---
[INFO] auto discovered modules [org.eclipse.hawkbit.ui.dd.CustomAcceptCriteria, org.eclipse.hawkbit.ui.rollout.groupschart.GroupsPieChart, org.eclipse.hawkbit.ui.customrenderers.CustomRendererWidgetSet, org.eclipse.hawkbit.ui.AppWidgetSet, org.eclipse.hawkbit.ui.filtermanagement.TextFieldSuggestionBox]
[INFO] 4 source files from GWT module org.eclipse.hawkbit.ui.dd.CustomAcceptCriteria
[INFO] 4 source files from GWT module org.eclipse.hawkbit.ui.rollout.groupschart.GroupsPieChart
[INFO] 11 source files from GWT module org.eclipse.hawkbit.ui.customrenderers.CustomRendererWidgetSet
[INFO] 1 source files from GWT module org.eclipse.hawkbit.ui.AppWidgetSet
[INFO] 7 source files from GWT module org.eclipse.hawkbit.ui.filtermanagement.TextFieldSuggestionBox
[INFO] 
[INFO] --- vaadin-maven-plugin:7.7.9:update-theme (default) @ hawkbit-ui ---
[INFO] Updating theme VAADIN\themes\hawkbit
[INFO] Widgetsets found from classpath:
.......
.......
.......
[INFO] --- vaadin-maven-plugin:7.7.9:compile-theme (default) @ hawkbit-ui ---
[INFO] Updating theme VAADIN\themes\hawkbit
[INFO] Theme "VAADIN\themes\hawkbit" compiled
[INFO] Updating theme VAADIN\themes\hawkbit
[INFO] Theme "VAADIN\themes\hawkbit" compiled
[INFO] Updating theme VAADIN\themes\hawkbit
[INFO] Theme "VAADIN\themes\hawkbit" compiled
```

If you have the `style.css` compiled and everything looks good and you still don't see the correct themeI would suggest cleaning browser caches, might help.

**Comment by keerthij** [comment]
I could find the widgestset and style.css generated. But I'm not able to access http://localhost:8080/VAADIN/themes/hawkbit/styles.css?v=7.7.9

As you mentioned, cleaned browser caches. still has no effect.

![image](https://user-images.githubusercontent.com/5600925/29265989-afdc9942-8100-11e7-807a-77b64c642ec8.png)

**Comment by michahirsch** [comment]
That's odd. How you are starting the hawkBit update-server app? From eclipse or using the `java -jar hawkbit-update-server-0.2.0-SNAPSHOT.jar` ? You are using the current master I guess, right?

**Comment by keerthij** [comment]
I'm running using java -jar command. Yes, I'm using current master.

**Comment by michahirsch** [comment]
Sorry, but I cannot reproduce this problem. I cleaned everything and build it from scratch and still working fine `java -jar hawkbit-update-server-0.2.0-SNAPSHOT.jar` . The style `style.css` exists and the UI is shown correctly. You are sure it's not a browser caching problem maybe?

**Comment by muellerml** [comment]
Had the same problem. Solution for me was to set the server.port to 8080 in the application.properties of the hawkbit-update-server.

**Comment by keerthij** [comment]
I'm not able to find server.port in application.properties of the hawkbit-update-server. Server.port is not present. Please help me to identify.

**Comment by michahirsch** [comment]
Actually the default `server.port` value is `8080`. So I'm not sure why setting it explicitly in the `application.properties` fixing the styling issues. Interesting if so! 

@keerthij  can you double check that the jar `hawkbit-ui-0.2.0-SNAPSHOT.jar` also contains the styles.css 
![image](https://user-images.githubusercontent.com/10206349/29561986-e19508e2-8737-11e7-9e5a-38ea75d85ce4.png)

**Comment by muellerml** [comment]
Actually, I now think it was just a coincidence that it suddenly started working for me when I added the parameter. Sometimes I still get the same behaviour as in the first post. 
Whenever I have a broken UI like @keerthij the request to the styles.css file returns a 404 error.

**Comment by djfun** [comment]
I have the same problem and have made the following observations:
When I clone the hawkbit repository and then execute `mvn clean install -DskipTests` the file hawkbit-ui/src/main/resources/VAADIN/themes/hawkbit/styles.css gets created but it doesn't get included in `hawkbit-ui-0.2.0-SNAPSHOT.jar`. As `mvn clean` does not seem to remove the created `styles.css`, when I run `mvn clean install -DskipTests` again, the `styles.css` from the previous run gets included.

**Comment by djfun** [comment]
The `styles.css` gets included when I remove line 45 from hawkbit-ui/pom.xml:
```diff
index 08ea6f50..5cdc4075 100644
--- a/hawkbit-ui/pom.xml
+++ b/hawkbit-ui/pom.xml
@@ -42,7 +42,6 @@
             </configuration>
             <executions>
                <execution>
-               <phase>process-classes</phase>
                   <configuration>
                      <!-- if you don't specify any modules, the plugin will find them -->
                      <!-- <modules> <module>com.vaadin.demo.mobilemail.gwt.ColorPickerWidgetSet</module> </modules> -->
```

**Comment by kaizimmerm** [comment]
Resolved by #587
